src/WebJobs.Script.WebHost/Diagnostics/DiagnosticEventInMemoryRepository.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Timers;
+using Microsoft.Extensions.Logging;
+using Timer = System.Timers.Timer;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Diagnostics
+{
+    public class DiagnosticEventInMemoryRepository : IDiagnosticEventRepository
+    {
+        private readonly ConcurrentDictionary<string, DiagnosticEvent> _events = new ConcurrentDictionary<string, DiagnosticEvent>();
+        private readonly Timer _resetTimer;
+
+        public DiagnosticEventInMemoryRepository()
+        {
+            _resetTimer = new Timer()
+            {
+                AutoReset = true,
+                Interval = 600000,
+                Enabled = true
+            };
+
+            _resetTimer.Elapsed += OnFlushLogs;
+        }
+
+        private void OnFlushLogs(object sender, ElapsedEventArgs e)
+        {
+            FlushLogs();
```

Comment:
This being an in-memory implementation of repository, FlushLogs would be same as purge as we would not retain anything for a long period in memory. Planning to send a repository code in a separate PR. To your point it would be a good idea to have an implementation of purge and keep Flush as on-op for inmemory repository.